tools/SetupFlow/DevHome.SetupFlow/Models/Environments/CreateEnvironmentTask.cs

```diff
@@ -0,0 +1,191 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+extern alias Projection;
+
+using System;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using CommunityToolkit.Mvvm.Messaging;
+using DevHome.Common.Environments.Models;
+using DevHome.Common.Environments.Services;
+using DevHome.Common.Models;
+using DevHome.SetupFlow.Models.Environments;
+using DevHome.SetupFlow.Services;
+using DevHome.SetupFlow.ViewModels;
+using Projection::DevHome.SetupFlow.ElevatedComponent;
+using Serilog;
+using Windows.Foundation;
+using DevHomeSDK = Microsoft.Windows.DevHome.SDK;
+
+namespace DevHome.SetupFlow.Models;
+
+/// <summary>
+/// Task that creates an environment using the user input from an adaptive card session.
+/// </summary>
+public sealed class CreateEnvironmentTask : ISetupTask, IDisposable, IRecipient<CreationAdaptiveCardSessionEndedMessage>
+{
+    private readonly ILogger _log = Log.ForContext("SourceContext", nameof(CreateEnvironmentTask));
+
+    private readonly IComputeSystemManager _computeSystemManager;
+
+    private readonly TaskMessages _taskMessages;
+
+    private readonly ActionCenterMessages _actionCenterMessages = new();
+
+    private readonly ISetupFlowStringResource _stringResource;
+
+    // Used to signal the task to start the creation operation. This is used when the adaptive card session ends.
+    private readonly AutoResetEvent _autoResetEventToStartCreationOperation = new(false);
+
+    private readonly SetupFlowViewModel _setupFlowViewModel;
+
+    private bool _disposedValue;
+
+    public event ISetupTask.ChangeMessageHandler AddMessage;
+
+    public string UserJsonInput { get; set; }
+
+    public ComputeSystemProviderDetails ProviderDetails { get; set; }
+
+    public DeveloperIdWrapper DeveloperIdWrapper { get; set; }
+
+#pragma warning disable 67
```

Comment:
Would be nice to have a comment to tell what warning 67 is about.

Reply:
updated thanks